Framework/API/src/FileProperty.cpp

```diff
@@ -56,21 +55,105 @@ void addExtension(const std::string &extension,
   else
     extensions.push_back(extension);
 }
+
+/**
+ * Get the path to the user's home directory (associated with ~) if it is set
+ * as an environment variable, and cache it
+ * @return The user's home path
+ */
+const std::string &getHomePath() {
+  static std::string homePath;
+  static bool initialised(false);
+
+  if (initialised) {
+    return homePath;
+  }
+  initialised = true;
+
+  char *home = std::getenv("HOME"); // Usually set on Windows and UNIX
+  if (home) {
+    homePath = std::string(home);
+    return homePath;
+  }
+
+  char *userProfile = std::getenv("USERPROFILE"); // Not usually set on UNIX
+  // Return even if it's an empty string, as we can do no better
+  homePath = userProfile ? std::string(userProfile) : "";
```

Comment:
Isn't the nullptr access also an issue for `homePath = std::string(home);` on the line above?